src/DIRAC/WorkloadManagementSystem/Executor/Scouting.py

```diff
@@ -0,0 +1,134 @@
+""" Executor to set status "Scouting" for a main job which has scout jobs
+"""
+
+__RCSID__ = "$Id: $"
+
+from DIRAC import S_OK, S_ERROR
+
+from DIRAC.WorkloadManagementSystem.Executor.Base.OptimizerExecutor import OptimizerExecutor
+from DIRAC.ResourceStatusSystem.Client.SiteStatus import SiteStatus
+from DIRAC.WorkloadManagementSystem.DB.JobDB import JobDB
+
+
+class Scouting(OptimizerExecutor):
+    """
+      The specific Optimizer must provide the following methods:
+      - optimizeJob() - the main method called for each job
+      and it can provide:
+      - initializeOptimizer() before each execution cycle
+    """
+
+    @classmethod
+    def initializeOptimizer(cls):
+        """ Initialization of the optimizer.
+        """
+        cls.siteClient = SiteStatus()
+        cls.__jobDB = JobDB()
+        return S_OK()
+
+    def optimizeJob(self, jid, jobState):
+        self.jobLog.info('Getting scoutparams from JobParameters')
+
+        result = self.__jobDB.getJobParameters(jid, ['ScoutFlag', 'ScoutID'])
+        if not result['OK']:
+            return S_ERROR('Could not retrieve scoutparams')
+
+        rCounter = 0
+        if result['Value']:
+            scoutparams = result['Value'].get(jid)
+            self.jobLog.info('scoutparams: %s' % scoutparams)
+            if not scoutparams:
+                self.jobLog.info('Skipping optimizer, since scoutparams are abnormal')
+                return self.setNextOptimizer(jobState)
+
+            scoutID, scoutFlag = self.__getIDandFlag(scoutparams)
+            if not scoutID:
+                self.jobLog.info('Skipping optimizers, since this job has not enough scoutparams.')
+                return self.setNextOptimizer(jobState)
+
+        else:
+            result = jobState.getManifest()
+            if not result['OK']:
+                return S_ERROR('Could not retrieve job manifest: %s' % result['Message'])
+            jobManifest = result['Value']
+            scoutID = jobManifest.getOption('ScoutID', None)
+            if not scoutID:
+                self.jobLog.info('Skipping optimizer, since no scout \
+                                 corresponding to this job group')
+                return self.setNextOptimizer(jobState)
+
+            scoutFlag = 0
+            result = jobState.getAttribute('RescheduleCounter')
+            if not result['OK']:
+                return S_ERROR('Could not retrieve RescheduleCounter')
+
+            rCounter = result['Value']
+            if int(rCounter) > 0:
```

Comment:
Recently we observed quite rare error there...line 61 gave 'Value' = None and the module crashed at line 66.
When I checked corresponding JobID, those jobs did not exist...probably deleted.
As far as I checked this error happens only once during past 2 years. 

Can we simply return S_ERROR here if we find None?

Reply:
Yes.